### PR TITLE
ci: add AzureLinux, rdma and ARM64 release test scenarios for cilium dual stack

### PR DESCRIPTION
**Pull request by camrynl**
**Reason for Change**:
Test Cilium dual stack with AzureLinux, rdma, and ARM64 and change daemonset for dual stack tests

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

File: .pipelines/cni/cilium/cilium-overlay-load-test-template.yaml [hunk 1]
```diff
@@ -93,7 +93,14 @@ stages:
                   kubectl apply -f test/integration/manifests/cilium/v${DIR}/cilium-agent/files
                   kubectl apply -f test/integration/manifests/cilium/v${DIR}/cilium-operator/files
 
-                  envsubst '${CILIUM_VERSION_TAG},${CILIUM_IMAGE_REGISTRY}' < test/integration/manifests/cilium/v${DIR}/cilium-agent/templates/daemonset.yaml | kubectl apply -f -
+                  if [ ! -z ${{ parameters.dualstackVersion }} ]; then
```
Comment (jpayne3506) [inline .pipelines/cni/cilium/cilium-overlay-load-test-template.yaml:96]: I believe the `!` needs to be outside of the `[ ]` to function properly

File: .pipelines/cni/pipeline.yaml [hunk 2]
```diff
@@ -359,6 +370,40 @@ stages:
...
+      dualstackVersion: ${CILIUM_DUALSTACK_VERSION}
+      cni: "cilium_dualstack"
+  
+  - template: cilium/cilium-overlay-load-test-template.yaml
+    parameters:
+      name: cilium_ds_rdma
+      clusterType: dualstack-byocni-nokubeproxy-up
+      clusterName: "cil-ds-rdma"
+      nodeCount: ${NODE_COUNT_CILIUM}
+      vmSize: Standard_HC44-16rs
```
Comment (jpayne3506) [inline .pipelines/cni/pipeline.yaml:403]: We might be getting close to our RDMA quota limit... Need to keep an eye on cluster creation after this is merged.

File: .pipelines/cni/pipeline.yaml [hunk 3]
```diff
@@ -359,6 +370,40 @@ stages:
...
+      osSKU: AzureLinux
+      dualstackVersion: ${CILIUM_DUALSTACK_VERSION}
+      cni: "cilium_dualstack"
+  
+  - template: cilium/cilium-overlay-load-test-template.yaml
+    parameters:
+      name: cilium_ds_rdma
+      clusterType: dualstack-byocni-nokubeproxy-up
+      clusterName: "cil-ds-rdma"
+      nodeCount: ${NODE_COUNT_CILIUM}
```
Comment (jpayne3506) [inline .pipelines/cni/pipeline.yaml:402]: This should be limited to 2.

**Review by jpayne3506 — CHANGES_REQUESTED** [review]

File: .pipelines/cni/pipeline.yaml (hunk 3 above)
Comment (camrynl) [reply .pipelines/cni/pipeline.yaml]: updated

File: .pipelines/cni/cilium/cilium-overlay-load-test-template.yaml (hunk 1 above)
Comment (camrynl) [reply .pipelines/cni/cilium/cilium-overlay-load-test-template.yaml:96]: This order also works, here's output from the pipeline: 
```
+ '[' '!' -z 1.14.10 ']'
+ echo 'Use dualstack daemonset for Cilium'
Use dualstack daemonset for Cilium
++ make ipv6-hp-bpf-version
```

**Review by jpayne3506 — APPROVED** [review]

**Pull request merged**